packages/api-client-core/src/types.ts

```diff
@@ -52,10 +59,10 @@ export type Select<Schema, Selection extends FieldSelection | null | undefined>
   ? Select<T, Selection>[]
   : Schema extends null
   ? Select<Exclude<Schema, null>, Selection> | null
-  : FilterNever<{
```

Comment:
If we remove this, there is a window of time where the generated API client for an app (potentially) uses this version of `api-client-core` but doesn't `DeepFilterNever` the return.

I think maybe we take a different approach: what if we give the current `Select` a new name, say `InnerSelect`, and instead do this:

```ts
export type Select<Schema, Selection extends FieldSelection | null | undefined> = DeepFilterNever<InnerSelect<Schema, Selection>>
```

which doesn't require a client rebuild and would only need a minor version bump (since it's backwards compatible).

Also, since the `Select` above has no constraints on the `Schema` type, we either need to consider adding the `extends Record<string, unknown>` constraint or making `DeepFilterNever` not need it, e.g.

```ts
export type DeepFilterNever<T> = T extends Record<string, unknown>
  ? FilterNever<{
      [Key in keyof T]:
         T[Key] extends Record<string, unknown>
         ? DeepFilterNever<T[Key]>
         : T[Key];
    }>
  : T
}>;
```

I think putting the constraint on `Schema` should be fine since a schema is never a literal / array / non-object thingy.

What do you think?

Reply:
And note, by "potentially uses", I think it would require the users to manually specify an `api-client-core` that would result in a `yarn install` grabbing the new version. Unlikely, but it _could_ happen. If there's any concern there, we can double check existing `package.json`s to see if this would happen.

Reply:
@thegedge that makes sense. One other thing that @airhorns brought up was that we don't `yarn install` on every rebuild so even though we've updated their `api-client-core` version the old version is the one that's imported until someone runs `yarn install`. So I think in addition to your suggested change, I'm going to re-implement `Select` in the same way in our codegen and remove it once all packages have been updated.